### PR TITLE
Resolved issues #696, #721, #722

### PR DESCRIPTION
**Pull request by WeaxMe**
Resolved issues #696, #721, #722

### CODE REVIEW

**Comment by WeaxMe** [comment]
- [ ] Remove unnecessary module https://github.com/WeaxMe/Orienteer/blob/develop/orienteer-core/src/main/java/org/orienteer/core/module/RoutingModule.java

- [ ] Refactor mappers in https://github.com/WeaxMe/Orienteer/tree/develop/orienteer-pages/src/main/java/org/orienteer/pages

- [ ] Add possibility for remove mapper https://github.com/WeaxMe/Orienteer/blob/develop/orienteer-pages/src/main/java/org/orienteer/pages/wicket/mapper/ODocumentAliasCompoundMapper.java

**Comment by WeaxMe** [comment]
### orienteer-users issues

- [ ] Add possibility for register user throughout  social networks.
Right now if user is not registered in system and he clicks on login throughout social networks user account creates by default which confuses users.

File: orienteer-users/src/main/java/org/orienteer/users/util/LoginError.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.orienteer.users.util;
+
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.ResourceModel;
+
+/**
+ * Login errors
+ */
+public enum LoginError {
```
Comment (PhantomYdn) [inline orienteer-users/src/main/java/org/orienteer/users/util/LoginError.java]: How this enum is being used?

File: orienteer-users/src/main/java/org/orienteer/users/util/RegistrationError.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package org.orienteer.users.util;
+
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.ResourceModel;
+
+/**
+ * Registration errors
+ */
+public enum RegistrationError {
```
Comment (PhantomYdn) [inline orienteer-users/src/main/java/org/orienteer/users/util/RegistrationError.java]: The same - why enum?

File: orienteer-core/src/main/java/org/orienteer/core/component/visualizer/UIVisualizersRegistry.java [hunk 3]
```diff
@@ -70,7 +70,11 @@ public void registerUIComponentFactory(IVisualizer visualizer)
 			registryTable.put(oType, visualizer.getName(), visualizer);
 		}
 	}
-	
+
+	public void unregisterUIComponentFactory(List<OType> types, String name) {
```
[the remark this answers is not in the document]
Comment (WeaxMe) [reply orienteer-core/src/main/java/org/orienteer/core/component/visualizer/UIVisualizersRegistry.java]: This need for unregister UI factory when custom module deactivated.

File: orienteer-core/src/main/java/org/orienteer/core/component/widget/AbstractCalculatedDocumentsWidget.java [hunk 4]
```diff
@@ -38,29 +39,40 @@ public AbstractCalculatedDocumentsWidget(String id, IModel<T> model,
...
-        		tablePanel = new GenericTablePanel<>("tablePanel",  new ResourceModel("error.class.not.defined"));
-        	}
-        } else {
-	        tablePanel = new GenericTablePanel<>("tablePanel",  new ResourceModel("error.query.not.defined"));
-        }
-        add(tablePanel);
+	}
+
+	@Override
+	protected void onInitialize() {
```
[the remark this answers is not in the document]
Comment (WeaxMe) [reply orienteer-core/src/main/java/org/orienteer/core/component/widget/AbstractCalculatedDocumentsWidget.java]: I think that it is more good: use constructor for initialize class fields and use onInitialize for initialize Wicket components

File: orienteer-pages/src/main/java/org/orienteer/pages/wicket/mapper/AbstractODocumentAliasMapper.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
+package org.orienteer.pages.wicket.mapper;
```
[the remark this answers is not in the document]
Comment (WeaxMe) [reply orienteer-pages/src/main/java/org/orienteer/pages/wicket/mapper/AbstractODocumentAliasMapper.java]: No this is not complex. But orienteer-pages looks like module for build CMS and custom URL feature looks like CMS feature, not core.

File: orienteer-test.properties [hunk 6]
```diff
@@ -1,7 +1,7 @@
 orienteer.production=true
 orienteer.application=org.orienteer.testenv.TestEnvOrienteerWebApplication
 orientdb.embedded=true
-orientdb.url=plocal:Orienteer
+orientdb.url=memory:Orienteer
```
[the remark this answers is not in the document]
Comment (WeaxMe) [reply orienteer-test.properties]: Test Orinteer in memory. No local folder with database for tests.

File: orienteer-users/src/main/java/org/orienteer/users/component/OUsersLoginPanel.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    public OUsersLoginPanel(String id, IModel<List<OAuth2Service>> services) {
+        super(id);
+        this.services = services;
+    }
+
+    @Override
+    protected void onInitialize() {
+        super.onInitialize();
+        add(new SocialNetworkPanel("socialNetworks", "panel.login.social.networks.title", services));
```
[the remark this answers is not in the document]
Comment (WeaxMe) [reply orienteer-users/src/main/java/org/orienteer/users/component/OUsersLoginPanel.java]: Discussed. Will add this feature

File: orienteer-users/src/main/java/org/orienteer/users/repository/DatabaseHelper.java [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+    private static final int ATTEMPTS = 10;
+
+    private static final String SELECT_TEMPLATE         = "select from %s where %s = ?";
+    private static final String DELETE_TEMPLATE         = "delete from %s where %s = ?";
+    private static final String DELETE_RECORDS_TEMPLATE = "delete from ?";
+
+    private DatabaseHelper() {}
+
+    public static void save(ODocumentWrapper wrapper) {
```
[the remark this answers is not in the document]
Comment (WeaxMe) [reply orienteer-users/src/main/java/org/orienteer/users/repository/DatabaseHelper.java]: This helper tries to save record up to 10 times and this prevents different versions of record error.
When first error occurs record reloads to last version and try to save again

File: orienteer-users/src/main/java/org/orienteer/users/util/LoginError.java (hunk 1 above)
Comment (WeaxMe) [reply orienteer-users/src/main/java/org/orienteer/users/util/LoginError.java]: This enum uses for login errors labels

File: orienteer-users/src/main/java/org/orienteer/users/util/RegistrationError.java (hunk 2 above)
Comment (WeaxMe) [reply orienteer-users/src/main/java/org/orienteer/users/util/RegistrationError.java]: This enum uses for registration errors labels

Comment (WeaxMe) [reply orienteer-users/src/main/java/org/orienteer/users/util/RegistrationError.java]: Additionally: I think in future if we will build more complex system this enum can be useful for indicates different errors

**Comment by WeaxMe** [comment]
- [ ] Use OrientDB tokens for login user

- [ ] Add flag which gives possibility for register user in login page by social networks

- [ ] Fix font in login page

- [ ] Add support for VK, GitLab, Twitter, LinkedIn

**Comment by WeaxMe** [comment]
Changed font in login and register pages

Screenshot_20190503_115251

Screenshot_20190503_115302

**Pull request merged**